x/distribution/types/genesis.go

```diff
@@ -67,3 +71,26 @@ func DefaultGenesisWithValidators(valAddrs []sdk.ValAddress) GenesisState {
 		DelegationDistInfos: ddis,
 	}
 }
+
+// ValidateGenesis validates the genesis state of distribution genesis input
+func ValidateGenesis(data GenesisState) error {
+	if data.CommunityTax.IsNegative() || data.CommunityTax.GT(sdk.OneDec()) {
+		return fmt.Errorf("mint parameter CommunityTax should be positive "+
+			"and less than one, is %s", data.CommunityTax.String())
+	}
+	if data.BaseProposerReward.IsNegative() {
+		return fmt.Errorf("mint parameter BaseProposerReward should be positive, is %s",
+			data.BaseProposerReward.String())
+	}
+	if data.BonusProposerReward.IsNegative() {
+		return fmt.Errorf("mint parameter BonusProposerReward should be positive, is %s",
+			data.BonusProposerReward.String())
+	}
+	if (data.BaseProposerReward.Add(data.BonusProposerReward)).
+		GT(sdk.OneDec()) {
+		return fmt.Errorf("mint parameters BaseProposerReward and "+
+			"BonusProposerReward cannot add to be greater than one be less than one, "+
```

Comment:
This sentence doesn't make too much sense, you just mean "cannot add to be greater than one", right?

Reply:
lol yeah, corrected